services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/MessageManagerImpl.java

```diff
@@ -673,6 +670,12 @@ public void sendPasswordChangeConfirmationEmail(long userId){
 
 		String messageBody = EmailUtils.readMailTemplate(MESSAGE_TEMPLATE_PASSWORD_CHANGE_CONFIRMATION, fieldValues);
 		String email = getEmailForUser(userId);
+		
+		if (emailQuarantineDao.isQuarantined(email)) {
+			logQuarantinedAddress("password change confirmation", email);
```

Comment:
throw exception here as sending an email is the last "security" check for a hacked account.